### PR TITLE
Some more drive-by WebGL fixes and cleanups

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21523)

### CODE REVIEW

**Comment by nox** [comment]
Disabled many more tests, notably everything related to texture sources that we don't support at all.

@bors-servo try

**Comment by nox** [comment]
Only unrelated failures again.

**Review by avadacatavra — CHANGES_REQUESTED** [review]

File: tests/wpt/webgl/meta/conformance/textures/canvas/__dir__.ini [hunk 1]
```diff
@@ -0,0 +1 @@
+disabled: canvas textures are not implemented yet
```
[the remark this answers is not in the document]
Comment (nox) [reply tests/wpt/webgl/meta/conformance/textures/canvas/__dir__.ini]: I was actually completely wrong about this specific folder of tests.

**Review by nox — COMMENTED** [review]
I still need to add a few issue links but it's gym time.

File: tests/wpt/webgl/meta/conformance/textures/misc/tex-video-using-tex-unit-non-zero.html.ini [hunk 2]
```diff
@@ -1,8 +1,3 @@
 [tex-video-using-tex-unit-non-zero.html]
-  expected: ERROR
-  [WebGL test #0: Unable to fetch WebGL rendering context for Canvas]
-    expected: FAIL
-
-  [Overall test]
-    expected: NOTRUN
+  disabled: video textures are not supported yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/misc/tex-video-using-tex-unit-non-zero.html.ini]: file and link to issue

File: tests/wpt/webgl/meta/conformance/textures/misc/texture-corner-case-videos.html.ini [hunk 3]
```diff
@@ -1,5 +1,3 @@
 [texture-corner-case-videos.html]
-  expected: TIMEOUT
-  [Overall test]
-    expected: NOTRUN
+  disabled: video textures are not supported yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/misc/texture-corner-case-videos.html.ini]: link to issue

File: tests/wpt/webgl/meta/conformance/textures/misc/texture-npot-video.html.ini [hunk 4]
```diff
@@ -1,5 +1,3 @@
 [texture-npot-video.html]
-  expected: TIMEOUT
-  [Overall test]
-    expected: NOTRUN
+  disabled: video textures are not supported yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/misc/texture-npot-video.html.ini]: link to issue

File: tests/wpt/webgl/meta/conformance/textures/svg_image/__dir__.ini [hunk 5]
```diff
@@ -0,0 +1 @@
+disabled: SVG support is not implemented yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/svg_image/__dir__.ini]: iirc we don't have plans to implement this right now, so if that's the case I don't see the need to file an issue for it

File: tests/wpt/webgl/meta/conformance/textures/video/__dir__.ini [hunk 6]
```diff
@@ -0,0 +1 @@
+disabled: video textures are not implemented yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/video/__dir__.ini]: link to issue

File: tests/wpt/webgl/meta/conformance/textures/webgl_canvas/__dir__.ini [hunk 7]
```diff
@@ -0,0 +1 @@
+disabled: canvas textures are not implemented yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/textures/webgl_canvas/__dir__.ini]: file/link to issue

File: tests/wpt/webgl/meta/conformance/canvas/draw-static-webgl-to-multiple-canvas-test.html.ini [hunk 8]
```diff
@@ -1,5 +1,3 @@
 [draw-static-webgl-to-multiple-canvas-test.html]
-  expected: ERROR
-  [Overall test]
-    expected: NOTRUN
+  disabled: drawing webgl canvas in 2d canvas is not supported yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/canvas/draw-static-webgl-to-multiple-canvas-test.html.ini]: link to issue

File: tests/wpt/webgl/meta/conformance/canvas/draw-webgl-to-canvas-test.html.ini [hunk 9]
```diff
@@ -1,5 +1,3 @@
 [draw-webgl-to-canvas-test.html]
-  expected: ERROR
-  [Overall test]
-    expected: NOTRUN
+  disabled: drawing webgl canvas in 2d canvas is not supported yet
```
Comment (avadacatavra) [inline tests/wpt/webgl/meta/conformance/canvas/draw-webgl-to-canvas-test.html.ini]: link to issue

**Comment by nox** [comment]
@jdm You need to sign on two things: disabling the quickcheck test and deleting the Servo-specific WebGL tests. @avadacatavra reviewed everything else.

File: tests/wpt/webgl/tests/js/js-test-pre.js [hunk 10]
```diff
@@ -143,8 +143,10 @@ function notifyFinishedToHarness() {
 
 (function () {
   var oldNotify = notifyFinishedToHarness;
+  var oldOnError = window.onerror;
   var t = async_test("Overall test");
   window.notifyFinishedToHarness = t.step_func_done(oldNotify);
+  window.onerror = t.step_func_done(oldNotify);
```
Comment (jdm) [inline tests/wpt/webgl/tests/js/js-test-pre.js]: This doesn't use oldOnError.

**Review by jdm — CHANGES_REQUESTED** [review]

File: tests/wpt/webgl/tests/js/js-test-pre.js (hunk 10 above)
Comment (nox) [reply tests/wpt/webgl/tests/js/js-test-pre.js]: I used `addEventListener` instead.

**Comment by jdm** [comment]
@bors-servo retry
* infra problems

**Comment by jdm** [comment]
@bors-servo retry
* busted build machine

**Pull request merged**